File: results/runs/pilot_pair_02_B/workspace/src/todos/todo_service.py
from datetime import datetime

class TODOItemService:
    def __init__(self, db_session):
        self.db_session = db_session

    def create_todo(self, user_id, description, reminder_datetime=None):
        new_todo = {
            'user_id': user_id,
            'description': description,
            'is_complete': False,
            'created_at': datetime.now(),
            'reminder_datetime': reminder_datetime
        }
        # Persist new_todo to the database
        self.db_session.add(new_todo)
        self.db_session.commit()
        return new_todo

    def list_todos(self, user_id):
        # Retrieve todos from database
        return self.db_session.query(TODOItem).filter_by(user_id=user_id).all()

    def complete_todo(self, todo_id):
        # Mark todo as complete in the database
        todo = self.db_session.query(TODOItem).filter_by(id=todo_id).first()
        todo.is_complete = True
        self.db_session.commit()
        return todo

    def delete_todo(self, todo_id):
        # Remove todo from the database
        todo = self.db_session.query(TODOItem).filter_by(id=todo_id).first()
        self.db_session.delete(todo)
        self.db_session.commit()

File: results/runs/pilot_pair_05_B/workspace/src/todos/todo.service.js
const db = require('../db');

async function createTodo(data) {
  const { userId, title, description, reminderDatetime } = data;
  const result = await db.query(
    'INSERT INTO todos (user_id, title, description, reminder_datetime) VALUES ($1, $2, $3, $4) RETURNING *',
    [userId, title, description, reminderDatetime]
  );
  return result.rows[0];
}

async function listTodos(userId) {
  const result = await db.query('SELECT * FROM todos WHERE user_id = $1', [userId]);
  return result.rows;
}

async function deleteTodo(todoId) {
  await db.query('DELETE FROM todos WHERE id = $1', [todoId]);
}

module.exports = {
  createTodo,
  listTodos,
  deleteTodo
};
